### ISSUE TITLE
Sequence Inference TypeError

### ISSUE THREAD

**Issue by jwustl** [jbohnslav/deepethogram#46]
Hi Jim, 

I'm setting up a project following the Getting Started guide and I'm getting an error when I run my sequence inference for the first time. I've attached the error log: 
[sequenceinferenceerror.txt](https://github.com/jbohnslav/deepethogram/files/6322387/sequenceinferenceerror.txt)

The flow generator, feature extractor and inference all ran successfully and it outputted predicted behaviors. I was wondering if you had any idea how to fix this error.

Thanks!

**Comment by jbohnslav** [comment]
This is a weird error indeed! Could you send me the HDF5 file ` C:/demo11_deepethogram\DATA\00023R1leftp2trim1-1\00023R1leftp2trim1-1_outputs.h5`?

**Comment by JBChaz** [comment]
Hi,

I have the exact same error message. This would be my HDF5 file:

https://drive.google.com/file/d/1S7lIivwHNw_AQ-gCLPtl3KdxatxWUBKK/view?usp=sharing

Thanks for any help

**Comment by jbohnslav** [comment]
Please upgrade to [the latest version of DeepEthogram](https://github.com/jbohnslav/deepethogram/blob/master/docs/beta.md) and see if this problem persists.

**Comment by JBChaz** [comment]
Upgraded to the latest version now I have new errors.
First problem I encountered was when clicking on the sequence inference button in the GUI:
```
[2021-05-27 16:25:05,781] INFO [deepethogram.projects.import_outputfile:774] Key used to load outputfile: resnet18
[2021-05-27 16:25:05,812] INFO [deepethogram.gui.main.import_outputfile:869] CHANGING LATENT NAME TO : resnet18
Traceback (most recent call last):
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\deepethogram\gui\main.py", line 565, in sequence_infer
    args = self.generate_sequence_inference_args()
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\deepethogram\gui\main.py", line 510, in generate_sequence_inference_args
    if sequence_weights is not None and os.path.isfile(sequence_weights):
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\genericpath.py", line 30, in isfile
    st = os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not dict
```
I think I fixed the issue by changing line 509 in gui/main.py from:
`sequence_weights = self.get_selected_models('sequence')`
to:
`sequence_weights = self.get_selected_models('sequence')['sequence']`.
After that the window to select the videos on which to run inference opened and the program continued.

However after that I ran into another problem, again related to the h5 files:

```
[2021-05-27 16:17:45,615] INFO [deepethogram.utils.load_state:339] loading from checkpoint file C:\Users\oteiza_lab\Documents\IRT2\First_steps_drinking_deepethogram - Copy\models\210525_115840_sequence_train_None\checkpoint.pt...
[2021-05-27 16:17:45,617] WARNING [deepethogram.utils.load_state_from_dict:244] classify1.weight not found in model dictionary
[2021-05-27 16:17:45,618] WARNING [deepethogram.utils.load_state_from_dict:244] classify1.bias not found in model dictionary
[2021-05-27 16:17:45,619] WARNING [deepethogram.utils.load_state_from_dict:244] classify2.weight not found in model dictionary
[2021-05-27 16:17:45,619] WARNING [deepethogram.utils.load_state_from_dict:244] classify2.bias not found in model dictionary
[2021-05-27 16:17:45,621] INFO [__main__.sequence_inference:294] best epoch from loaded file: -1
Traceback (most recent call last):
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\deepethogram\sequence\inference.py", line 297, in sequence_inference
    thresholds = f['val']['metrics_by_threshold']['optimum'][best_epoch, :]
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\h5py\_hl\group.py", line 288, in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py\h5o.pyx", line 190, in h5py.h5o.open
KeyError: "Unable to open object (object 'metrics_by_threshold' doesn't exist)"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\deepethogram\sequence\inference.py", line 332, in <module>
    sequence_inference(cfg)
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\deepethogram\sequence\inference.py", line 300, in sequence_inference
    thresholds = f['threshold_curves']['val']['optimum'][best_epoch, :]
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\h5py\_hl\dataset.py", line 777, in __getitem__
    selection = sel.select(self.shape, args, dataset=self)
  File "C:\Users\oteiza_lab\Anaconda3\envs\DEG\lib\site-packages\h5py\_hl\selections.py", line 82, in select
    return selector.make_selection(args)
  File "h5py\_selector.pyx", line 272, in h5py._selector.Selector.make_selection
  File "h5py\_selector.pyx", line 107, in h5py._selector.Selector.apply_args
ValueError: 2 indexing arguments for 1 dimensions

```
I don't know how to continue from there. Any Idea what could have gone wrong? Let me know if I should open another issue for this.

Thank you very much for all the help.

**Comment by jbohnslav** [comment]
I was able to reproduce the GUI button bug. It should be fixed in b1932be3a71eb49b192f4dd7ba18b98d9ae056d2. Please `pip install --upgrade deepethogram` to 0.1.1 to see if that fixes things. 

The sequence inference bug was fixed in 5794544372832ae4ef41071a2240ed9e94815edf. Note: you should probably re-train your sequence models with DeepEthogram 0.1, in which case you wouldn't have this ValueError. 

Feel free to re-open if this didn't fix your issue.
